index/file_category.go

```diff
@@ -0,0 +1,104 @@
+package index
+
+import (
+	"errors"
+
+	"github.com/go-enry/go-enry/v2"
+)
+
+// FileCategory represents the category of a file, as determined by go-enry. It is non-exhaustive
+// but tries to the major cases like whether the file is a test, generated, etc.
+//
+// A file's category is used in search scoring to determine the weight of a file match.
+type FileCategory byte
+
+const (
+	// FileCategoryMissing is a sentinel value that indicates we never computed the file category during indexing
+	// (which means we're reading from an old index version). This value can never be written to the index.
+	FileCategoryMissing FileCategory = iota
+	FileCategoryDefault
+	FileCategoryTest
+	FileCategoryVendored
+	FileCategoryGenerated
+	FileCategoryConfig
+	FileCategoryDotFile
+	FileCategoryDocumentation
+)
+
+func DetermineFileCategory(doc *Document) {
+	name := doc.Name
+	content := doc.Content
+
+	// If this document has been skipped, it's likely very large. In this case, we just guess the category based
+	// on the filename to avoid examining the contents. Note: passing nil content is allowed by the go-enry contract.
+	if doc.SkipReason != "" {
+		content = nil
```

Comment:
I think this might break our little hack to replace content with NOT-INDEXED if we skipped a file 🤔

Reply:
What do you mean? To clarify, I am assigning the local `content` field here, not `doc.Content`.

Reply:
🤦 never mind ;-)